### DISCUSSION TITLE
User folder setup

### DISCUSSION THREAD

**Discussion by noootch** [Akkudoktor-EOS/EOS#155] [Polls]
Hello all,

in https://github.com/Akkudoktor-EOS/EOS/pull/141 I suggest a change from config.py to a json config.

In the PR I have two possible solutions to define a user directory:
1. via command line
2. via env var

The problem I see in this is that if someone uses both variants, sooner or later there might be an issue, when changing the directory in one of these methods.

I would like to reduce the options to either CLI args or ENVVAR.

What are your opinions on this?

**Comment by b0661** [comment]
Make CLI overwrite ENVVAR.

**Comment by drbacke** [comment]
I'm Not Sure If this ist helpfull. The Server should Run in a docker env, so starting the py Code directly will be for devs only

**Reply by noootch** [reply]
good point. Then I guess envar(s) would be enough. Especially when run in a container. CLI is some effort for a clean implementation.

**Comment by eckelj** [comment]
I suggest using https://pypi.org/project/python-decouple/ it's perfect for several settings. env vars but also config files like .env or similar. 

it gives the flexibility that you usually need.
